cnd/src/network/comit_ln.rs

```diff
@@ -577,3 +577,281 @@ impl NetworkBehaviourEventProcess<oneshot_behaviour::OutEvent<finalize::Message>
+        Swarm::listen_on(&mut bob_swarm, bob_addr.clone())
+            .with_context(|| format!("Address is not supported: {:?}", bob_addr))
+            .unwrap();
+
+        let send_swap_digest = random_swap_digest();
+
+        let dial_info = DialInformation {
+            peer_id: bob_peer_id.clone(),
+            address_hint: Some(bob_addr.clone()),
+        };
+
+        let ethereum_id = EthereumIdentity::from(Address::default());
+
+        let secp_pubkey = secp256k1::PublicKey::from_str(
+            "02c2a8efce029526d364c2cf39d89e3cdda05e5df7b2cbfc098b4e3d02b70b5275",
+        )
+        .unwrap();
+        let lightning_id = crate::lightning::PublicKey::from(secp_pubkey);
+
+        let swap_params_alice = HanEtherereumHalightBitcoinCreateSwapParams {
+            role: Role::Alice,
+            peer: dial_info.clone(),
+            ethereum_identity: ethereum_id,
+            ethereum_absolute_expiry: Timestamp::from(10),
+            ethereum_amount: asset::Ether::zero(),
+            lightning_identity: lightning_id,
+            lightning_cltv_expiry: Timestamp::from(10),
+            lightning_amount: asset::Lightning::from_sat(0),
+        };
+
+        let swap_params_bob = HanEtherereumHalightBitcoinCreateSwapParams {
+            role: Role::Bob,
+            peer: dial_info, /* not relevant for Bob, should be optional or somehow related
+                              * to role */
+            ethereum_identity: ethereum_id,
+            ethereum_absolute_expiry: Timestamp::from(10),
+            ethereum_amount: asset::Ether::zero(),
+            lightning_identity: lightning_id,
+            lightning_cltv_expiry: Timestamp::from(10),
+            lightning_amount: asset::Lightning::from_sat(0),
+        };
+
+        let ethereum_identity_bob: Address = swap_params_bob.ethereum_identity.clone().into();
+        let ethereum_identity_alice: Address = swap_params_alice.ethereum_identity.clone().into();
+
+        // let alice_swarm_arc = Arc::new(alice_swarm);
+        // let bob_swarm_arc = Arc::new(bob_swarm);
+
+        let alice_swarm = Arc::new(Mutex::new(alice_swarm));
+        let bob_swarm = Arc::new(Mutex::new(bob_swarm));
+
+        // construct future to asynchroniously kick off the swap communication for alice
+        let future_init = start_communication(
+            alice_swarm.clone(),
+            bob_swarm.clone(),
+            swap_params_alice,
+            swap_params_bob,
+            NodeLocalSwapId::default(),
+            NodeLocalSwapId::default(),
+        );
+
+        // future to poll for the finalized event of alice behaviour
+        let alice_future = async {
+            println!("Start listening for events for alice");
+            loop {
+                // sleep(time::Duration::from_millis(300));
+
+                let mut guard = alice_swarm.lock().await;
+
+                let next = guard.next_event().await;
+
```

Comment:
Are you sure its ok to do this?  I don't think one can hold a lock while calling `await`?